File: samples/java/DomainObject/src/com/alachisoft/nosdb/entityobject/Order.java
// ===============================================================================
// Alachisoft (R) NosDB Sample Code.
// ===============================================================================
// Copyright © Alachisoft.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
// ===============================================================================

package com.nosdb.entityobject;

import java.util.ArrayList;
import java.util.Date;

public class Order
{
    public String OrderID;
    public Date OrderDate;
    public Date RequiredDate;
    public Date ShippedDate;
    public double Freight;
    public String ShipName;
    public String ShipAddress;
    public String ShipCity;
    public String ShipRegion;
    public String ShipPostalCode;
    public String ShipCountry;
    public String CustomerID;
    public String EmployeeID;
    public String ShipperID;
    public ArrayList<OrderDetail> OrderDetails;

}

File: samples/dotnet/EntityObjects/EntityObjects/Order.cs
// ===============================================================================
// Alachisoft (R) NosDB Sample Code.
// ===============================================================================
// Copyright © Alachisoft.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
// ===============================================================================

using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace NosDB.Samples.EntityObjects
{
    public class Order
    {
        [JsonProperty(PropertyName = "OrderID")]
        public long ID { get; set; }

        public DateTime OrderDate { get; set; }
        public DateTime RequiredDate { get; set; }
        public DateTime ShippedDate { get; set; }

        public double Freight { get; set; }
        public string ShipName { get; set; }
        public string ShipAddress { get; set; }
        public string ShipCity { get; set; }
        public string ShipRegion { get; set; }
        public string ShipPostalCode { get; set; }
        public string ShipCountry { get; set; }

        public string CustomerID { get; set; }
        public long EmployeeID { get; set; }
        public string ShipperID { get; set; }

        public List<OrderDetail> OrderDetails { get; set; }
    }
}
